packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -218,15 +227,20 @@ function OrgSelect({
     });
   };
 
-  const filterDataAsAdmin = debounce(() => {
-    const mapResponseToOrgs = (response: Array<OrgFilterResponse>) =>
-      response.map(({ groups, ...org }) => org);
-    fetchOrgsAsAdmin(value)
-      .then(response => response.json())
-      .then(mapResponseToOrgs)
-      .then(setFilteredData)
-      .catch(console.error);
-  }, 300);
+  const filterDataAsAdmin = useCallback(
```

Comment:
Without useCallback the debounce was being redefined on every component render (aka every time `value` changed). Such that if you typed in `test` you'd get 4 definitions of `filterDataAsAdmin` with each ultimately being invoked after their debounce time elapsed.